src/core/namespace.c

```diff
@@ -658,9 +674,17 @@ static int mount_private_dev(MountEntry *m) {
                         goto fail;
                 }
         } else {
-                r = clone_device_node("/dev/ptmx", temporary_mount);
-                if (r < 0)
-                        goto fail;
+                r = clone_device_node("/dev/ptmx", temporary_mount, can_mknod);
+                if (r != -EPERM) {
```

Comment:
no sure i follow the logic here? aren't we supposed to revert to non-device-node behaviour precisely when we get EPERM? You implemented it the opposite way, no?

Also, if the boolean is already unset we don't need this fallback, do we?

Reply:
ugh, addressed

Reply:
but the boolean will always be set to true here